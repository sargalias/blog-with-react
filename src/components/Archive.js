import React from 'react';
import Container from './Container';
import BlogListings from './BlogListings';

const Archive = ({ blogPosts }) => (
  <section className="blog mt-5">
    <Container>
      <BlogListings blogPosts={blogPosts} />
    </Container>
  </section>
);

export default Archive;
