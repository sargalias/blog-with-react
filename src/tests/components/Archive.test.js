import React from 'react';
import { shallow } from 'enzyme';
import Archive from '../../components/Archive';
import blogPosts from '../fixtures/blogPosts';

test('Archive should render correctly with no blog posts', () => {
  const wrapper = shallow(<Archive />);
  expect(wrapper).toMatchSnapshot();
});

test('Archive should render correctly with blog posts', () => {
  const wrapper = shallow(<Archive blogPosts={blogPosts} />);
  expect(wrapper).toMatchSnapshot();
});

test('Archive calls BlogListings correctly with blog posts', () => {
  const wrapper = shallow(<Archive blogPosts={blogPosts} />);
  const blogListings = wrapper.find('BlogListings');
  expect(blogListings.props().blogPosts).toEqual(blogPosts);
});

test('Archive calls blogListings correctly with no blog posts', () => {
  const wrapper = shallow(<Archive />);
  const blogListings = wrapper.find('BlogListings');
  expect(blogListings.props().blogPosts).toBe(undefined);
});
