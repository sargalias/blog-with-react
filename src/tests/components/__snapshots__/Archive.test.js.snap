// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Archive should render correctly with blog posts 1`] = `
<section
  className="blog mt-5"
>
  <Container>
    <BlogListings
      blogPosts={
        Array [
          Object {
            "category": "Food",
            "date": "1970-01-01T00:00:00.000Z",
            "excerpt": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Est fugiat harum obcaecati pariatur sint tempore unde. Aperiam commodi distinctio earum eligendi, enim itaque laudantium magni neque porro temporibus vitae voluptatum?",
            "imageURL": "https://via.placeholder.com/350x350",
            "postId": "1",
            "title": "Test title",
            "user": Object {
              "displayName": "User displayName 1",
              "id": 1,
            },
          },
          Object {
            "category": "Travel",
            "date": "1970-01-06T00:00:00.000Z",
            "excerpt": "Test excerpt 2",
            "imageURL": "https://via.placeholder.com/200x200",
            "postId": "2",
            "title": "Test title 2",
            "user": Object {
              "displayName": "User displayName 2",
              "id": 2,
            },
          },
          Object {
            "category": "Food",
            "date": "1969-12-27T00:00:00.000Z",
            "excerpt": "3 Lorem ipsum dolor",
            "imageURL": "https://via.placeholder.com/250x250",
            "postId": "3",
            "title": "Test title 3",
            "user": Object {
              "displayName": "User displayName 3",
              "id": 3,
            },
          },
        ]
      }
    />
  </Container>
</section>
`;

exports[`Archive should render correctly with no blog posts 1`] = `
<section
  className="blog mt-5"
>
  <Container>
    <BlogListings />
  </Container>
</section>
`;
